src/main/java/edu/uwaterloo/ece658/service/S3Service.java

```diff
@@ -0,0 +1,111 @@
+        try {
+                java.util.Date expiration = new java.util.Date();
+                long milliSeconds = expiration.getTime();
+                milliSeconds += 1000 * 60 * 60; // Add 1 hour.
+                expiration.setTime(milliSeconds);
+
+                GeneratePresignedUrlRequest generatePresignedUrlRequest = 
+                            new GeneratePresignedUrlRequest(bucketName, objectKey);
+                generatePresignedUrlRequest.setMethod(method); 
+                generatePresignedUrlRequest.setExpiration(expiration);
+
+                url = s3Client.generatePresignedUrl(generatePresignedUrlRequest); 
+        } catch (AmazonServiceException exception) {
+                System.out.println("Caught an AmazonServiceException, " +
+                    "which means your request made it to Amazon S3, " +
+                    "but was rejected with an error response for some reason.");
+                System.out.println("Error Message: " + exception.getMessage());
+                System.out.println("HTTP  Code: "    + exception.getStatusCode());
+                System.out.println("AWS Error Code:" + exception.getErrorCode());
+                System.out.println("Error Type:    " + exception.getErrorType());
+                System.out.println("Request ID:    " + exception.getRequestId());
+        } catch (AmazonClientException ace) {
+                System.out.println("Caught an AmazonClientException, " +
+                    "which means the client encountered an internal error " +
+                    "while trying to communicate with S3, such as not being " +
+                    "able to access the network.");
+                System.out.println("Error Message: " + ace.getMessage());
+        }
+        return url;            
+    }
+    
+    public boolean deleteImage(String s3key) {
+        BasicAWSCredentials creds =
+                new BasicAWSCredentials(accessKey, secretKey);
+        
+        AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withCredentials(
+                new AWSStaticCredentialsProvider(creds)).build();
+        
+        try {
+            s3Client.deleteObject(new DeleteObjectRequest(bucketName, s3key));
+        } catch (AmazonServiceException ase) {
+            System.out.println("Caught an AmazonServiceException.");
+            System.out.println("Error Message:    " + ase.getMessage());
+            System.out.println("HTTP Status Code: " + ase.getStatusCode());
+            System.out.println("AWS Error Code:   " + ase.getErrorCode());
+            System.out.println("Error Type:       " + ase.getErrorType());
+            System.out.println("Request ID:       " + ase.getRequestId());
+            return false;
+        } catch (AmazonClientException ace) {
+            System.out.println("Caught an AmazonClientException.");
+            System.out.println("Error Message: " + ace.getMessage());
+            return false;
+        }
+        return true;

```

Comment:
If exception happened, it still returns true, meaning photo was successfully deleted.